src/App.tsx

```diff
@@ -8,37 +8,114 @@ import { PostsList } from './components/PostsList';
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [errorGetComments, setErrorGetComments] = useState(false);
+  const [activeForm, setActiveForm] = useState<Post | null>(null);
+
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(() => 'Error get Users');
+  }, []);
+
+  useEffect(() => {
+    setErrorGetPosts(false);
+    if (selectedUser) {
+      setIsLoadingPosts(true);
+      getPosts(selectedUser.id)
+        .then(gettedPosts => {
+          setComments([]);
+          setSelectedPost(null);
+          setPosts(gettedPosts);
+        })
+        .catch(() => {
+          setErrorGetPosts(true);
+        })
+        .finally(() => setIsLoadingPosts(false));
+    }
+  }, [selectedUser]);
+
+  useEffect(() => {
+    setErrorGetComments(false);
+    setActiveForm(null);
+    if (selectedPost) {
+      setIsLoadingComments(true);
+      getComments(selectedPost.id)
+        .then(setComments)
+        .catch(() => {
+          setErrorGetComments(true);
+        })
+        .finally(() => setIsLoadingComments(false));
+    }
+  }, [selectedPost]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  selectedUser={selectedUser}
+                  setSelectedUser={setSelectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
-
-                <Loader />
-
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {!selectedUser ? (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                ) : (
+                  <>
+                    {isLoadingPosts ? (
+                      <Loader />
+                    ) : (
+                      <>
+                        {errorGetPosts && (
+                          <div
+                            className="notification is-danger"
+                            data-cy="PostsLoadingError"
+                          >
+                            Something went wrong!
+                          </div>
+                        )}
```

Comment:
Nested ternary looks too complicated, try to split this logic to improve code readibility